_episodes/01-why-bother-performance.md

```diff
@@ -1,16 +1,133 @@
+- "Just because a way is correct doesn't always mean it is the best option"
 ---
 
-FIXME
+## What is software performance?
 
-{% include links.md %}
 
+
+## Why is software performance important?
+
+(FIXME)
+
+> ## Enchancing performance: rationale
+>
+> Imagine you had a 40 x 40 x 40 box like the one below, divided up into smaller boxes, each measuring 1 x 1 x 1. If you wanted and you wanted to simulate what was happening inside each smaller box for 10 hours each How long will the example below take to run on one core?
+>
+> include figure.html url="" max-width="40%" file="/fig/01/40x40x40_cube.png" alt="Figure" caption="40 x 40 x 40 cube"
+> 
+> {: language-bash}
+> 
+>> ## Solution
+>> 
+>> 64000 hours or just over 73 years.
+>>
+>> This is way longer than anyone could bear! But remember, that is utilising just one core. If you had a machine that could simulate each of those smaller boxes simultaneously and a code that enables each box to effectively interact with each other, the whole job would only take roughly 10-12 hours. So it provides the motivation we need to speed up the code we have and utilise the machine to its best capacity
+>>
+>{: .solution}
+{: .challenge}
+
+## How is performance measured?
+
+There are a number of key terms in computing when it comes to understanding performance and expected duration of tasks. The most important of these are flops, walltime and cpuh
+
+> ### Flops
+>
+> Flops stands for floating point operations per second and they are typically used to measure the performance of a computer's processor
+>
+{: .callout}
+
+
+> ### Walltime
+> 
+> Walltime is simply the length of time, usually measured in seconds that a program takes to run or to execute its assigned tasks. 
+>
+> This is specified in a submission script when before your task officially starts running. If the task takes longer than the walltime that is specified, the task gets killed and will not complete.
+>
+{: .callout}
+
+> ### CPUH
+>
+> CPU hours is the processor time needed to complete a task, i.e. the amount of computing work done. 
+>
+>
+{: .callout}
+
+
+> ## Calculate CPU hours
+> 
+> In the following example, assume that you are utilising all the available core in a node. Calculate the CPU hours required to run the following job. Each node has 40 cores.
+> 
+> ```
+> #SBATCH --nodes = 2
+> #SBATCH --time = 05:00:00
+> ```
+> {: .bash}
+> 
+> > ## Solution
+> >
+> > 400 CPUh.
+> >
+> > 
+> > 
+>{: .solution}
+{: .challenge}
+
+## How can performance be enhanced?
+
+(FIXME) Mention pool of workers, tuning, OpenMP, MPI, GPU and libraries and will refer to them later
+
+## HPC layout and partitioning
+
+> ## Enhancing Performance
+>
+> Enhacing performance is one of the most important things in computation. Which of these are viable ways of enhancing performance? There may be more than one correct answer.
+>
+> 1. Utilising more cores
+> 2. Utilising more nodes
+> 3. Increasing the simulation walltime and CPUh
+> 4. Optimising the code
```

Comment:
Before this we might need to explain to the users about optimisation. Probable blidspot.

Reply:
Inserted section, which we can add to later.